workspaces-tests/src/bin/upgrade.rs

```diff
@@ -0,0 +1,97 @@
+#![allow(missing_docs)]
+
+// Ignore
+pub fn main() {}
+
+use std::fmt::Display;
+
+use near_contract_tools::{
+    rbac::Rbac,
+    slot::Slot,
+    upgrade::{
+        upgrade::{self, Upgrade},
+        Upgrade, UpgradeHook,
+    },
+    Rbac,
+};
+use near_sdk::{
+    borsh::{self, BorshDeserialize, BorshSerialize},
+    env, near_bindgen, AccountId, BorshStorageKey, PanicOnDefault,
+};
+use thiserror::Error;
+
+#[derive(BorshSerialize, BorshStorageKey)]
+enum StorageKey {
+    Upgrade,
+}
+
+#[derive(PanicOnDefault, BorshSerialize, BorshDeserialize, Upgrade)]
+#[near_bindgen]
+pub struct Contract {
+    uint32: var1,
+}
+
+#[derive(PanicOnDefault, BorshSerialize, BorshDeserialize, Upgrade)]
+#[near_bindgen]
+pub struct ContractV2 {
+    uint32: var2,
+}
+
+
+// This single function implementation completely implements simple multisig on
+// the contract
+impl Upgrade for Contract {
+    fn upgrade() {
+        env::log_str("Upgrading contract");
+        Contract::upgrade();
+    }
+}
+
+impl UpgradeHook for Contract {
+    fn on_upgrade(&mut self) {
+        env::log_str("Upgraded!");
```

Comment:
I was also unable to find this string in the logs. Does this function get called?